server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/RollbackPlanRuntimeFlowableTask.java

```diff
@@ -201,38 +196,46 @@ protected RollbackPlanTaskResult start(Long taskId, TaskService taskService, Del
 
     private RollbackPlanTaskResult handleRollbackResult(String rollbackResult) {
         if (StringUtils.isNotBlank(rollbackResult)) {
-            String objectId =
-                    putRollbackPlan(rollbackResult, flowTaskProperties.getMaxRollbackContentSizeBytes());
-            this.objectId = objectId;
-            this.isSuccess = true;
-            return RollbackPlanTaskResult.success(this.objectId);
+            long totalSizeBytes = rollbackResult.getBytes().length;
+            long maxSizeBytes = rollbackProperties.getMaxRollbackContentSizeBytes();
+            if (totalSizeBytes > maxSizeBytes) {
+                log.warn("Rollback plan result file size exceeds maximum, totalSize={}, taskId={}", totalSizeBytes,
+                        getTaskId());
+                throw new UnsupportedSqlTypeForRollbackPlanException(
+                        "Rollback plan result file size exceeds maximum, totalSize=" + totalSizeBytes
+                                + " Byte, max size=" + maxSizeBytes + " Byte");
+            }
+            try {
+                String resultFileDownloadUrl = String.format(
+                        "/api/v2/flow/flowInstances/%s/tasks/rollbackPlan/download", getFlowInstanceId());
+                String resultFileRootPath = FileManager.generatePath(FileBucket.ROLLBACK_PLAN);
+                String resultFileId = StringUtils.uuid();
+                String filePath = String.format("%s/%s.sql", resultFileRootPath, resultFileId);
+                FileUtils.writeStringToFile(new File(filePath), rollbackResult, StandardCharsets.UTF_8);
+                if (Objects.nonNull(cloudObjectStorageService) && cloudObjectStorageService.supported()) {
+                    File tempFile = new File(filePath);
+                    try {
+                        String objectName = cloudObjectStorageService.uploadTemp(resultFileId + ".sql", tempFile);
+                        resultFileDownloadUrl = cloudObjectStorageService.getBucketName() + "/" + objectName;
+                        log.info("Upload generated rollback plan task result file to OSS, file name={}", resultFileId);
+                    } finally {
+                        OdcFileUtil.deleteFiles(tempFile);
+                    }
+                }
+                this.resultFileId = resultFileId;
+                this.resultFileDownloadUrl = resultFileDownloadUrl;
+                this.isSuccess = true;
+                return RollbackPlanTaskResult.success(resultFileId, resultFileDownloadUrl);
+            } catch (Exception e) {
+                log.warn("Failed to put generated rollback plan file for taskId={}", getTaskId());
```

Comment:
It is recommended to print the stack information of the exception to the log to facilitate locating the problem.

Reply:
Yes you are right, I fix it.